src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -108,9 +110,9 @@ public String getDisplayName() {
 
         public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project) {
             // You should have permission to configure your project in order to get the available managed files
-            if (project != null) {
-                project.checkPermission(Item.CONFIGURE);
-            }
+            Permission permission = context == null ? Item.CREATE : Item.CONFIGURE;
+            AccessControlled ac = project == null ? Jenkins.get() : project;
+            ac.checkPermission(permission);
```

Comment:
The `context` is null when the pipeline syntax editor is opened from http://localhost:8080/jenkins/pipeline-syntax .  I believe that would cause a null pointer exception when the proposed code is executed.

Reply:
@MarkEWaite just to double check you are referring to `project` or `context`   Everything should have `Jenkins` as the stapler root object IIUC and Jenkins is an `ItemGroup` so `context` should never be `null` but `project` may?

and running in the debugger shows

![image](https://user-images.githubusercontent.com/494726/142222869-0c44a97a-e663-4c63-ad8a-6f13facf5c96.png)